drivers/hmis_external_apis/spec/models/hmis_external_apis/ac_hmis/exporters/cde_export_spec.rb

```diff
@@ -46,15 +46,21 @@
   end
 
   context 'when there are multiple CDEs with different owners and CDEDs' do
-    let!(:hud_service_2) { create :hmis_hud_service, data_source: ds, client: c1, enrollment: e1 }
-    let!(:cded2) { create :hmis_custom_data_element_definition, label: 'A different CDED', data_source: ds, owner_type: 'Hmis::Hud::Service', field_type: :string }
-    let!(:cde2) { create :hmis_custom_data_element, data_element_definition: cded2, owner: hud_service_2, data_source: ds, value_string: 'A new value' }
+    let!(:records) do
+      cdes = []
+      10.times do
+        service = create(:hmis_hud_service, data_source: ds, client: c1, enrollment: e1)
+        cded = create(:hmis_custom_data_element_definition, label: 'A different CDED', data_source: ds, owner_type: 'Hmis::Hud::Service', field_type: :string)
+        cdes << create(:hmis_custom_data_element, data_element_definition: cded, owner: service, data_source: ds, value_string: 'A new value')
+      end
+      cdes
+    end
 
     it 'does not do a db lookup per iteration' do
       expect do
         subject.run!
-        expect(subject.send(:cdes).length).to eq(2)
-      end.to make_database_queries(count: 4)
+        expect(subject.send(:cdes).length).to eq(11)
+      end.to make_database_queries(count: 5..8)
```

Comment:
Updated this test to make it a bit more flexible. It was failing because I added a new query.